lib/archethic/mining/validation_context.ex

```diff
@@ -800,17 +800,17 @@ defmodule Archethic.Mining.ValidationContext do
     protocol_version = Mining.protocol_version()
 
     ops =
-      %LedgerValidation{fee: fee}
+      %LedgerValidation{fee: fee, transaction_movements: movements}
```

Comment:
I think to keep a workflow like previously and to avoid errors, I think the movements could be passed in arguments to the function `validate_sufficient_funds` which will add them in the struct. Also we could set the `transaction_movements` field of the LedgerValidation struct therefore the `build_resolved_movements` function could have a guard to ensure the movements is a list. This could avoid doing misstake in ordering the functions (You did not update the order in TransactionFactory and no error is raised)